feat(website): add macos amd64 desktop

alexfreska opened the pull request:
- The desktop downloads now include AMD64 for macOS.

alexfreska commented:
* **#786** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/786?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alexfreska and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

n8maninger commented:
Looks like a problem with the GitHub release. We are downloading the `.zip` instead of the `.dmg`. The `.dmg` does not include the architecture so amd64/arm64 are probably overwriting each other.

alexfreska commented:
@n8maninger I've added dmgs for both architectures over on desktop: https://github.com/SiaFoundation/desktop/pull/96 and updated this PR to match those changes.
